src/cpx_solve.jl

```diff
@@ -79,28 +80,53 @@ function get_objval(model::Model)
   end
   return objval[1]
 end
+get_objval = c_api_getobjval
 
-function get_solution(model::Model)
-  nvars = num_var(model)
-  x = Vector{Cdouble}(nvars)
+function c_api_solninfo(model::Model)    
+  solnmethod_p = [Cint(-1)]
```

Comment:
You can make these `solnmethod_p = Ref{Cint}()`, and then return `solnmethod_p[]`. This avoids allocating a vector.